Index: xbmc/lib/libPython/linux/Makefile.in
===================================================================
---  xbmc/lib/libPython/linux/Makefile.in.orig	2010-08-29 21:38:20.939294072 -0400
+++  xbmc/lib/libPython/linux/Makefile.in	2010-08-29 21:39:46.832294131 -0400
@@ -2,7 +2,10 @@
 
 SHELL=/bin/bash
 SYSDIR=../../../../system/python
-ifeq (@USE_PYTHON2_6@,1)
+ifeq (@USE_PYTHON2_7@,1)
+    PYVERSION=python2.7
+    SO=python27-$(ARCH).so
+else ifeq (@USE_PYTHON2_6@,1)
     PYVERSION=python2.6
     SO=python26-$(ARCH).so
 else ifeq (@USE_PYTHON2_5@,1)
Index: xbmc/lib/libPython/xbmcmodule/winxmldialog.cpp
===================================================================
--- xbmc/lib/libPython/xbmcmodule/winxmldialog.cpp	(revision 32022)
+++ xbmc/lib/libPython/xbmcmodule/winxmldialog.cpp	(working copy)
@@ -24,7 +24,9 @@
 #endif
 #include "winxml.h"
 #if (defined USE_EXTERNAL_PYTHON)
-  #if (defined HAVE_LIBPYTHON2_6)
+  #if (defined HAVE_LIBPYTHON2_7)
+    #include <python2.7/Python.h>
+  #elif (defined HAVE_LIBPYTHON2_6)
     #include <python2.6/Python.h>
   #elif (defined HAVE_LIBPYTHON2_5)
     #include <python2.5/Python.h>
Index: xbmc/lib/libPython/xbmcmodule/dialog.h
===================================================================
--- xbmc/lib/libPython/xbmcmodule/dialog.h	(revision 32022)
+++ xbmc/lib/libPython/xbmcmodule/dialog.h	(working copy)
@@ -23,7 +23,9 @@
   #include "config.h"
 #endif
 #if (defined USE_EXTERNAL_PYTHON)
-  #if (defined HAVE_LIBPYTHON2_6)
+  #if (defined HAVE_LIBPYTHON2_7)
+    #include <python2.7/Python.h>
+  #elif (defined HAVE_LIBPYTHON2_6)
     #include <python2.6/Python.h>
   #elif (defined HAVE_LIBPYTHON2_5)
     #include <python2.5/Python.h>
Index: xbmc/lib/libPython/xbmcmodule/xbmcguimodule.cpp
===================================================================
--- xbmc/lib/libPython/xbmcmodule/xbmcguimodule.cpp	(revision 32022)
+++ xbmc/lib/libPython/xbmcmodule/xbmcguimodule.cpp	(working copy)
@@ -23,7 +23,10 @@
   #include "config.h"
 #endif
 #if (defined USE_EXTERNAL_PYTHON)
-  #if (defined HAVE_LIBPYTHON2_6)
+  #if (defined HAVE_LIBPYTHON2_7)
+    #include <python2.7/Python.h>
+    #include <python2.7/structmember.h>
+  #elif (defined HAVE_LIBPYTHON2_6)
     #include <python2.6/Python.h>
     #include <python2.6/structmember.h>
   #elif (defined HAVE_LIBPYTHON2_5)
Index: xbmc/lib/libPython/xbmcmodule/keyboard.h
===================================================================
--- xbmc/lib/libPython/xbmcmodule/keyboard.h	(revision 32022)
+++ xbmc/lib/libPython/xbmcmodule/keyboard.h	(working copy)
@@ -23,7 +23,9 @@
   #include "config.h"
 #endif
 #if (defined USE_EXTERNAL_PYTHON)
-  #if (defined HAVE_LIBPYTHON2_6)
+  #if (defined HAVE_LIBPYTHON2_7)
+    #include <python2.7/Python.h>
+  #elif (defined HAVE_LIBPYTHON2_6)
     #include <python2.6/Python.h>
   #elif (defined HAVE_LIBPYTHON2_5)
     #include <python2.5/Python.h>
Index: xbmc/lib/libPython/xbmcmodule/pyplaylist.cpp
===================================================================
--- xbmc/lib/libPython/xbmcmodule/pyplaylist.cpp	(revision 32022)
+++ xbmc/lib/libPython/xbmcmodule/pyplaylist.cpp	(working copy)
@@ -26,7 +26,9 @@
 #include "Util.h"
 #include "pyplaylist.h"
 #if (defined USE_EXTERNAL_PYTHON)
-  #if (defined HAVE_LIBPYTHON2_6)
+  #if (defined HAVE_LIBPYTHON2_7)
+    #include <python2.7/Python.h>
+  #elif (defined HAVE_LIBPYTHON2_6)
     #include <python2.6/Python.h>
   #elif (defined HAVE_LIBPYTHON2_5)
     #include <python2.5/Python.h>
Index: xbmc/lib/libPython/xbmcmodule/controllist.cpp
===================================================================
--- xbmc/lib/libPython/xbmcmodule/controllist.cpp	(revision 32022)
+++ xbmc/lib/libPython/xbmcmodule/controllist.cpp	(working copy)
@@ -23,7 +23,9 @@
   #include "config.h"
 #endif
 #if (defined USE_EXTERNAL_PYTHON)
-  #if (defined HAVE_LIBPYTHON2_6)
+  #if (defined HAVE_LIBPYTHON2_7)
+    #include <python2.7/Python.h>
+  #elif (defined HAVE_LIBPYTHON2_6)
     #include <python2.6/Python.h>
   #elif (defined HAVE_LIBPYTHON2_5)
     #include <python2.5/Python.h>
Index: xbmc/lib/libPython/xbmcmodule/winxml.cpp
===================================================================
--- xbmc/lib/libPython/xbmcmodule/winxml.cpp	(revision 32022)
+++ xbmc/lib/libPython/xbmcmodule/winxml.cpp	(working copy)
@@ -24,7 +24,9 @@
 #endif
 #include "winxml.h"
 #if (defined USE_EXTERNAL_PYTHON)
-  #if (defined HAVE_LIBPYTHON2_6)
+  #if (defined HAVE_LIBPYTHON2_7)
+    #include <python2.7/Python.h>
+  #elif (defined HAVE_LIBPYTHON2_6)
     #include <python2.6/Python.h>
   #elif (defined HAVE_LIBPYTHON2_5)
     #include <python2.5/Python.h>
Index: xbmc/lib/libPython/xbmcmodule/controllabel.cpp
===================================================================
--- xbmc/lib/libPython/xbmcmodule/controllabel.cpp	(revision 32022)
+++ xbmc/lib/libPython/xbmcmodule/controllabel.cpp	(working copy)
@@ -23,7 +23,9 @@
   #include "config.h"
 #endif
 #if (defined USE_EXTERNAL_PYTHON)
-  #if (defined HAVE_LIBPYTHON2_6)
+  #if (defined HAVE_LIBPYTHON2_7)
+    #include <python2.7/Python.h>
+  #elif (defined HAVE_LIBPYTHON2_6)
     #include <python2.6/Python.h>
   #elif (defined HAVE_LIBPYTHON2_5)
     #include <python2.5/Python.h>
Index: xbmc/lib/libPython/xbmcmodule/action.h
===================================================================
--- xbmc/lib/libPython/xbmcmodule/action.h	(revision 32022)
+++ xbmc/lib/libPython/xbmcmodule/action.h	(working copy)
@@ -23,7 +23,9 @@
   #include "config.h"
 #endif
 #if (defined USE_EXTERNAL_PYTHON)
-  #if (defined HAVE_LIBPYTHON2_6)
+  #if (defined HAVE_LIBPYTHON2_7)
+    #include <python2.7/Python.h>
+  #elif (defined HAVE_LIBPYTHON2_6)
     #include <python2.6/Python.h>
   #elif (defined HAVE_LIBPYTHON2_5)
     #include <python2.5/Python.h>
Index: xbmc/lib/libPython/xbmcmodule/controlimage.cpp
===================================================================
--- xbmc/lib/libPython/xbmcmodule/controlimage.cpp	(revision 32022)
+++ xbmc/lib/libPython/xbmcmodule/controlimage.cpp	(working copy)
@@ -23,7 +23,9 @@
   #include "config.h"
 #endif
 #if (defined USE_EXTERNAL_PYTHON)
-  #if (defined HAVE_LIBPYTHON2_6)
+  #if (defined HAVE_LIBPYTHON2_7)
+    #include <python2.7/Python.h>
+  #elif (defined HAVE_LIBPYTHON2_6)
     #include <python2.6/Python.h>
   #elif (defined HAVE_LIBPYTHON2_5)
     #include <python2.5/Python.h>
Index: xbmc/lib/libPython/xbmcmodule/infotagvideo.h
===================================================================
--- xbmc/lib/libPython/xbmcmodule/infotagvideo.h	(revision 32022)
+++ xbmc/lib/libPython/xbmcmodule/infotagvideo.h	(working copy)
@@ -23,7 +23,9 @@
   #include "config.h"
 #endif
 #if (defined USE_EXTERNAL_PYTHON)
-  #if (defined HAVE_LIBPYTHON2_6)
+  #if (defined HAVE_LIBPYTHON2_7)
+    #include <python2.7/Python.h>
+  #elif (defined HAVE_LIBPYTHON2_6)
     #include <python2.6/Python.h>
   #elif (defined HAVE_LIBPYTHON2_5)
     #include <python2.5/Python.h>
Index: xbmc/lib/libPython/xbmcmodule/control.h
===================================================================
--- xbmc/lib/libPython/xbmcmodule/control.h	(revision 32022)
+++ xbmc/lib/libPython/xbmcmodule/control.h	(working copy)
@@ -23,7 +23,9 @@
   #include "config.h"
 #endif
 #if (defined USE_EXTERNAL_PYTHON)
-  #if (defined HAVE_LIBPYTHON2_6)
+  #if (defined HAVE_LIBPYTHON2_7)
+    #include <python2.7/Python.h>
+  #elif (defined HAVE_LIBPYTHON2_6)
     #include <python2.6/Python.h>
   #elif (defined HAVE_LIBPYTHON2_5)
     #include <python2.5/Python.h>
Index: xbmc/lib/libPython/xbmcmodule/controlradiobutton.cpp
===================================================================
--- xbmc/lib/libPython/xbmcmodule/controlradiobutton.cpp	(revision 32022)
+++ xbmc/lib/libPython/xbmcmodule/controlradiobutton.cpp	(working copy)
@@ -23,7 +23,9 @@
   #include "config.h"
 #endif
 #if (defined USE_EXTERNAL_PYTHON)
-  #if (defined HAVE_LIBPYTHON2_6)
+  #if (defined HAVE_LIBPYTHON2_7)
+    #include <python2.7/Python.h>
+  #elif (defined HAVE_LIBPYTHON2_6)
     #include <python2.6/Python.h>
   #elif (defined HAVE_LIBPYTHON2_5)
     #include <python2.5/Python.h>
Index: xbmc/lib/libPython/xbmcmodule/controlgroup.cpp
===================================================================
--- xbmc/lib/libPython/xbmcmodule/controlgroup.cpp	(revision 32022)
+++ xbmc/lib/libPython/xbmcmodule/controlgroup.cpp	(working copy)
@@ -23,7 +23,9 @@
   #include "config.h"
 #endif
 #if (defined USE_EXTERNAL_PYTHON)
-  #if (defined HAVE_LIBPYTHON2_6)
+  #if (defined HAVE_LIBPYTHON2_7)
+    #include <python2.7/Python.h>
+  #elif (defined HAVE_LIBPYTHON2_6)
     #include <python2.6/Python.h>
   #elif (defined HAVE_LIBPYTHON2_5)
     #include <python2.5/Python.h>
Index: xbmc/lib/libPython/xbmcmodule/infotagmusic.h
===================================================================
--- xbmc/lib/libPython/xbmcmodule/infotagmusic.h	(revision 32022)
+++ xbmc/lib/libPython/xbmcmodule/infotagmusic.h	(working copy)
@@ -23,7 +23,9 @@
   #include "config.h"
 #endif
 #if (defined USE_EXTERNAL_PYTHON)
-  #if (defined HAVE_LIBPYTHON2_6)
+  #if (defined HAVE_LIBPYTHON2_7)
+    #include <python2.7/Python.h>
+  #elif (defined HAVE_LIBPYTHON2_6)
     #include <python2.6/Python.h>
   #elif (defined HAVE_LIBPYTHON2_5)
     #include <python2.5/Python.h>
Index: xbmc/lib/libPython/xbmcmodule/controltextbox.cpp
===================================================================
--- xbmc/lib/libPython/xbmcmodule/controltextbox.cpp	(revision 32022)
+++ xbmc/lib/libPython/xbmcmodule/controltextbox.cpp	(working copy)
@@ -23,7 +23,9 @@
   #include "config.h"
 #endif
 #if (defined USE_EXTERNAL_PYTHON)
-  #if (defined HAVE_LIBPYTHON2_6)
+  #if (defined HAVE_LIBPYTHON2_7)
+    #include <python2.7/Python.h>
+  #elif (defined HAVE_LIBPYTHON2_6)
     #include <python2.6/Python.h>
   #elif (defined HAVE_LIBPYTHON2_5)
     #include <python2.5/Python.h>
Index: xbmc/lib/libPython/xbmcmodule/listitem.h
===================================================================
--- xbmc/lib/libPython/xbmcmodule/listitem.h	(revision 32022)
+++ xbmc/lib/libPython/xbmcmodule/listitem.h	(working copy)
@@ -25,7 +25,9 @@
   #include "config.h"
 #endif
 #if (defined USE_EXTERNAL_PYTHON)
-  #if (defined HAVE_LIBPYTHON2_6)
+  #if (defined HAVE_LIBPYTHON2_7)
+    #include <python2.7/Python.h>
+  #elif (defined HAVE_LIBPYTHON2_6)
     #include <python2.6/Python.h>
   #elif (defined HAVE_LIBPYTHON2_5)
     #include <python2.5/Python.h>
Index: xbmc/lib/libPython/xbmcmodule/xbmcmodule.cpp
===================================================================
--- xbmc/lib/libPython/xbmcmodule/xbmcmodule.cpp	(revision 32022)
+++ xbmc/lib/libPython/xbmcmodule/xbmcmodule.cpp	(working copy)
@@ -23,7 +23,9 @@
   #include "config.h"
 #endif
 #if (defined USE_EXTERNAL_PYTHON)
-  #if (defined HAVE_LIBPYTHON2_6)
+  #if (defined HAVE_LIBPYTHON2_7)
+    #include <python2.7/Python.h>
+  #elif (defined HAVE_LIBPYTHON2_6)
     #include <python2.6/Python.h>
   #elif (defined HAVE_LIBPYTHON2_5)
     #include <python2.5/Python.h>
Index: xbmc/lib/libPython/xbmcmodule/dialog.cpp
===================================================================
--- xbmc/lib/libPython/xbmcmodule/dialog.cpp	(revision 32022)
+++ xbmc/lib/libPython/xbmcmodule/dialog.cpp	(working copy)
@@ -24,7 +24,9 @@
 #endif
 #include "dialog.h"
 #if (defined USE_EXTERNAL_PYTHON)
-  #if (defined HAVE_LIBPYTHON2_6)
+  #if (defined HAVE_LIBPYTHON2_7)
+    #include <python2.7/Python.h>
+  #elif (defined HAVE_LIBPYTHON2_6)
     #include <python2.6/Python.h>
   #elif (defined HAVE_LIBPYTHON2_5)
     #include <python2.5/Python.h>
Index: xbmc/lib/libPython/xbmcmodule/controlfadelabel.cpp
===================================================================
--- xbmc/lib/libPython/xbmcmodule/controlfadelabel.cpp	(revision 32022)
+++ xbmc/lib/libPython/xbmcmodule/controlfadelabel.cpp	(working copy)
@@ -23,7 +23,9 @@
   #include "config.h"
 #endif
 #if (defined USE_EXTERNAL_PYTHON)
-  #if (defined HAVE_LIBPYTHON2_6)
+  #if (defined HAVE_LIBPYTHON2_7)
+    #include <python2.7/Python.h>
+  #elif (defined HAVE_LIBPYTHON2_6)
     #include <python2.6/Python.h>
   #elif (defined HAVE_LIBPYTHON2_5)
     #include <python2.5/Python.h>
Index: xbmc/lib/libPython/xbmcmodule/player.h
===================================================================
--- xbmc/lib/libPython/xbmcmodule/player.h	(revision 32022)
+++ xbmc/lib/libPython/xbmcmodule/player.h	(working copy)
@@ -23,7 +23,9 @@
   #include "config.h"
 #endif
 #if (defined USE_EXTERNAL_PYTHON)
-  #if (defined HAVE_LIBPYTHON2_6)
+  #if (defined HAVE_LIBPYTHON2_7)
+    #include <python2.7/Python.h>
+  #elif (defined HAVE_LIBPYTHON2_6)
     #include <python2.6/Python.h>
   #elif (defined HAVE_LIBPYTHON2_5)
     #include <python2.5/Python.h>
Index: xbmc/lib/libPython/xbmcmodule/PythonPlayer.h
===================================================================
--- xbmc/lib/libPython/xbmcmodule/PythonPlayer.h	(revision 32022)
+++ xbmc/lib/libPython/xbmcmodule/PythonPlayer.h	(working copy)
@@ -25,7 +25,9 @@
   #include "config.h"
 #endif
 #if (defined USE_EXTERNAL_PYTHON)
-  #if (defined HAVE_LIBPYTHON2_6)
+  #if (defined HAVE_LIBPYTHON2_7)
+    #include <python2.7/Python.h>
+  #elif (defined HAVE_LIBPYTHON2_6)
     #include <python2.6/Python.h>
   #elif (defined HAVE_LIBPYTHON2_5)
     #include <python2.5/Python.h>
Index: xbmc/lib/libPython/xbmcmodule/controlprogress.cpp
===================================================================
--- xbmc/lib/libPython/xbmcmodule/controlprogress.cpp	(revision 32022)
+++ xbmc/lib/libPython/xbmcmodule/controlprogress.cpp	(working copy)
@@ -23,7 +23,9 @@
   #include "config.h"
 #endif
 #if (defined USE_EXTERNAL_PYTHON)
-  #if (defined HAVE_LIBPYTHON2_6)
+  #if (defined HAVE_LIBPYTHON2_7)
+    #include <python2.7/Python.h>
+  #elif (defined HAVE_LIBPYTHON2_6)
     #include <python2.6/Python.h>
   #elif (defined HAVE_LIBPYTHON2_5)
     #include <python2.5/Python.h>
Index: xbmc/lib/libPython/xbmcmodule/controlspin.cpp
===================================================================
--- xbmc/lib/libPython/xbmcmodule/controlspin.cpp	(revision 32022)
+++ xbmc/lib/libPython/xbmcmodule/controlspin.cpp	(working copy)
@@ -23,7 +23,9 @@
   #include "config.h"
 #endif
 #if (defined USE_EXTERNAL_PYTHON)
-  #if (defined HAVE_LIBPYTHON2_6)
+  #if (defined HAVE_LIBPYTHON2_7)
+    #include <python2.7/Python.h>
+  #elif (defined HAVE_LIBPYTHON2_6)
     #include <python2.6/Python.h>
   #elif (defined HAVE_LIBPYTHON2_5)
     #include <python2.5/Python.h>
Index: xbmc/lib/libPython/xbmcmodule/controlbutton.cpp
===================================================================
--- xbmc/lib/libPython/xbmcmodule/controlbutton.cpp	(revision 32022)
+++ xbmc/lib/libPython/xbmcmodule/controlbutton.cpp	(working copy)
@@ -23,7 +23,9 @@
   #include "config.h"
 #endif
 #if (defined USE_EXTERNAL_PYTHON)
-  #if (defined HAVE_LIBPYTHON2_6)
+  #if (defined HAVE_LIBPYTHON2_7)
+    #include <python2.7/Python.h>
+  #elif (defined HAVE_LIBPYTHON2_6)
     #include <python2.6/Python.h>
   #elif (defined HAVE_LIBPYTHON2_5)
     #include <python2.5/Python.h>
Index: xbmc/lib/libPython/xbmcmodule/window.h
===================================================================
--- xbmc/lib/libPython/xbmcmodule/window.h	(revision 32022)
+++ xbmc/lib/libPython/xbmcmodule/window.h	(working copy)
@@ -23,7 +23,9 @@
   #include "config.h"
 #endif
 #if (defined USE_EXTERNAL_PYTHON)
-  #if (defined HAVE_LIBPYTHON2_6)
+  #if (defined HAVE_LIBPYTHON2_7)
+    #include <python2.7/Python.h>
+  #elif (defined HAVE_LIBPYTHON2_6)
     #include <python2.6/Python.h>
   #elif (defined HAVE_LIBPYTHON2_5)
     #include <python2.5/Python.h>
Index: xbmc/lib/libPython/xbmcmodule/pyplaylist.h
===================================================================
--- xbmc/lib/libPython/xbmcmodule/pyplaylist.h	(revision 32022)
+++ xbmc/lib/libPython/xbmcmodule/pyplaylist.h	(working copy)
@@ -25,7 +25,9 @@
   #include "config.h"
 #endif
 #if (defined USE_EXTERNAL_PYTHON)
-  #if (defined HAVE_LIBPYTHON2_6)
+  #if (defined HAVE_LIBPYTHON2_7)
+    #include <python2.7/Python.h>
+  #elif (defined HAVE_LIBPYTHON2_6)
     #include <python2.6/Python.h>
   #elif (defined HAVE_LIBPYTHON2_5)
     #include <python2.5/Python.h>
Index: xbmc/lib/libPython/xbmcmodule/GUIPythonWindow.h
===================================================================
--- xbmc/lib/libPython/xbmcmodule/GUIPythonWindow.h	(revision 32022)
+++ xbmc/lib/libPython/xbmcmodule/GUIPythonWindow.h	(working copy)
@@ -26,7 +26,9 @@
   #include "config.h"
 #endif
 #if (defined USE_EXTERNAL_PYTHON)
-  #if (defined HAVE_LIBPYTHON2_6)
+  #if (defined HAVE_LIBPYTHON2_7)
+    #include <python2.7/Python.h>
+  #elif (defined HAVE_LIBPYTHON2_6)
     #include <python2.6/Python.h>
   #elif (defined HAVE_LIBPYTHON2_5)
     #include <python2.5/Python.h>
Index: xbmc/lib/libPython/xbmcmodule/winxml.h
===================================================================
--- xbmc/lib/libPython/xbmcmodule/winxml.h	(revision 32022)
+++ xbmc/lib/libPython/xbmcmodule/winxml.h	(working copy)
@@ -23,7 +23,9 @@
   #include "config.h"
 #endif
 #if (defined USE_EXTERNAL_PYTHON)
-  #if (defined HAVE_LIBPYTHON2_6)
+  #if (defined HAVE_LIBPYTHON2_7)
+    #include <python2.7/Python.h>
+  #elif (defined HAVE_LIBPYTHON2_6)
     #include <python2.6/Python.h>
   #elif (defined HAVE_LIBPYTHON2_5)
     #include <python2.5/Python.h>
Index: xbmc/lib/libPython/xbmcmodule/controlcheckmark.cpp
===================================================================
--- xbmc/lib/libPython/xbmcmodule/controlcheckmark.cpp	(revision 32022)
+++ xbmc/lib/libPython/xbmcmodule/controlcheckmark.cpp	(working copy)
@@ -23,7 +23,9 @@
   #include "config.h"
 #endif
 #if (defined USE_EXTERNAL_PYTHON)
-  #if (defined HAVE_LIBPYTHON2_6)
+  #if (defined HAVE_LIBPYTHON2_7)
+    #include <python2.7/Python.h>
+  #elif (defined HAVE_LIBPYTHON2_6)
     #include <python2.6/Python.h>
   #elif (defined HAVE_LIBPYTHON2_5)
     #include <python2.5/Python.h>
Index: xbmc/lib/libPython/xbmcmodule/listitem.cpp
===================================================================
--- xbmc/lib/libPython/xbmcmodule/listitem.cpp	(revision 32022)
+++ xbmc/lib/libPython/xbmcmodule/listitem.cpp	(working copy)
@@ -23,7 +23,9 @@
   #include "config.h"
 #endif
 #if (defined USE_EXTERNAL_PYTHON)
-  #if (defined HAVE_LIBPYTHON2_6)
+  #if (defined HAVE_LIBPYTHON2_7)
+    #include <python2.7/Python.h>
+  #elif (defined HAVE_LIBPYTHON2_6)
     #include <python2.6/Python.h>
   #elif (defined HAVE_LIBPYTHON2_5)
     #include <python2.5/Python.h>
Index: xbmc/lib/libPython/xbmcmodule/pyutil.h
===================================================================
--- xbmc/lib/libPython/xbmcmodule/pyutil.h	(revision 32022)
+++ xbmc/lib/libPython/xbmcmodule/pyutil.h	(working copy)
@@ -25,7 +25,9 @@
   #include "config.h"
 #endif
 #if (defined USE_EXTERNAL_PYTHON)
-  #if (defined HAVE_LIBPYTHON2_6)
+  #if (defined HAVE_LIBPYTHON2_7)
+    #include <python2.7/Python.h>
+  #elif (defined HAVE_LIBPYTHON2_6)
     #include <python2.6/Python.h>
   #elif (defined HAVE_LIBPYTHON2_5)
     #include <python2.5/Python.h>
Index: xbmc/lib/libPython/xbmcmodule/language.h
===================================================================
--- xbmc/lib/libPython/xbmcmodule/language.h.orig	2010-08-30 02:53:36.545418596 -0400
+++ xbmc/lib/libPython/xbmcmodule/language.h	2010-08-30 02:54:07.381668564 -0400
@@ -25,7 +25,9 @@
   #include "config.h"
 #endif
 #if (defined USE_EXTERNAL_PYTHON)
-  #if (defined HAVE_LIBPYTHON2_6)
+  #if (defined HAVE_LIBPYTHON2_7)
+    #include <python2.7/Python.h>
+  #elif (defined HAVE_LIBPYTHON2_6)
     #include <python2.6/Python.h>
   #elif (defined HAVE_LIBPYTHON2_5)
     #include <python2.5/Python.h>
Index: xbmc/lib/libPython/xbmcmodule/PythonSettings.h
===================================================================
--- xbmc/lib/libPython/xbmcmodule/PythonSettings.h.orig	2010-08-30 03:02:55.131669137 -0400
+++ xbmc/lib/libPython/xbmcmodule/PythonSettings.h	2010-08-30 03:03:18.218294899 -0400
@@ -25,7 +25,9 @@
   #include "config.h"
 #endif
 #if (defined USE_EXTERNAL_PYTHON)
-  #if (defined HAVE_LIBPYTHON2_6)
+  #if (defined HAVE_LIBPYTHON2_7)
+    #include <python2.7/Python.h>
+  #elif (defined HAVE_LIBPYTHON2_6)
     #include <python2.6/Python.h>
   #elif (defined HAVE_LIBPYTHON2_5)
     #include <python2.5/Python.h>
Index: xbmc/lib/libPython/XBPyThread.h
===================================================================
--- xbmc/lib/libPython/XBPyThread.h	(revision 32022)
+++ xbmc/lib/libPython/XBPyThread.h	(working copy)
@@ -26,7 +26,9 @@
   #include "config.h"
 #endif
 #if (defined USE_EXTERNAL_PYTHON)
-  #if (defined HAVE_LIBPYTHON2_6)
+  #if (defined HAVE_LIBPYTHON2_7)
+    #include <python2.7/Python.h>
+  #elif (defined HAVE_LIBPYTHON2_6)
     #include <python2.6/Python.h>
   #elif (defined HAVE_LIBPYTHON2_5)
     #include <python2.5/Python.h>
Index: xbmc/lib/libPython/XBPyThread.cpp
===================================================================
--- xbmc/lib/libPython/XBPyThread.cpp	(revision 32022)
+++ xbmc/lib/libPython/XBPyThread.cpp	(working copy)
@@ -24,7 +24,10 @@
   #include "config.h"
 #endif
 #if (defined USE_EXTERNAL_PYTHON)
-  #if (defined HAVE_LIBPYTHON2_6)
+  #if (defined HAVE_LIBPYTHON2_7)
+    #include <python2.7/Python.h>
+    #include <python2.7/osdefs.h>
+  #elif (defined HAVE_LIBPYTHON2_6)
     #include <python2.6/Python.h>
     #include <python2.6/osdefs.h>
   #elif (defined HAVE_LIBPYTHON2_5)
Index: xbmc/lib/libPython/XBPython.cpp
===================================================================
--- xbmc/lib/libPython/XBPython.cpp	(revision 32022)
+++ xbmc/lib/libPython/XBPython.cpp	(working copy)
@@ -24,7 +24,9 @@
   #include "config.h"
 #endif
 #if (defined USE_EXTERNAL_PYTHON)
-  #if (defined HAVE_LIBPYTHON2_6)
+  #if (defined HAVE_LIBPYTHON2_7)
+    #include <python2.7/Python.h>
+  #elif (defined HAVE_LIBPYTHON2_6)
     #include <python2.6/Python.h>
   #elif (defined HAVE_LIBPYTHON2_5)
     #include <python2.5/Python.h>
Index: configure.in
===================================================================
--- configure.in	(revision 32022)
+++ configure.in	(working copy)
@@ -636,18 +636,21 @@
 
 # External Python
 if test "$use_external_libraries" = "yes" || test "$use_external_python" = "yes"; then
-  AC_CHECK_LIB([python2.6], [main],,
+  AC_CHECK_LIB([python2.7], [main],,
+  [AC_CHECK_LIB([python2.6], [main],,
   [AC_CHECK_LIB([python2.5], [main],,
   [AC_CHECK_LIB([python2.4], [main],,
-  [AC_MSG_ERROR($missing_library)] )] )] )
+  [AC_MSG_ERROR($missing_library)] )] )] )] )
 
   # Test again, but this time just set some variables we need in Makefiles
-  AC_CHECK_LIB([python2.6], [main],[USE_PYTHON2_6=1],
+  AC_CHECK_LIB([python2.7], [main],[USE_PYTHON2_7=1],
+  [AC_CHECK_LIB([python2.6], [main],[USE_PYTHON2_6=1],
   [AC_CHECK_LIB([python2.5], [main],[USE_PYTHON2_5=1],
   [AC_CHECK_LIB([python2.4], [main],[USE_PYTHON2_4=1],
-  [AC_MSG_ERROR($missing_library)] )] )] )
+  [AC_MSG_ERROR($missing_library)] )] )] )] )
 
   AC_MSG_NOTICE($external_python_enabled)
+  test "$USE_PYTHON2_7" && AC_MSG_NOTICE([Using Python 2.7])
   test "$USE_PYTHON2_6" && AC_MSG_NOTICE([Using Python 2.6])
   test "$USE_PYTHON2_5" && AC_MSG_NOTICE([Using Python 2.5])
   test "$USE_PYTHON2_4" && AC_MSG_NOTICE([Using Python 2.4])
@@ -1032,6 +1035,7 @@
 AC_SUBST(USE_EXTERNAL_LIBOGG)
 AC_SUBST(USE_EXTERNAL_LIBWAVPACK)
 AC_SUBST(USE_EXTERNAL_PYTHON)
+AC_SUBST(USE_PYTHON2_7)
 AC_SUBST(USE_PYTHON2_6)
 AC_SUBST(USE_PYTHON2_5)
 AC_SUBST(USE_PYTHON2_4)
